From 64de223c37307a85a878abd384dcadf3750bfcd2 Mon Sep 17 00:00:00 2001
From: Hecheng Yu <kde-yyds@qq.com>
Date: Sat, 1 Aug 2026 14:03:10 +0800
Subject: [PATCH 4/8] dodge: Clamp 8K capture rate to 30fps to fix 7.5fps
 recording

After the 4K120 -> 8K switch, the recorder is configured with captureRate=120
while the encoder outputs at 30fps (log: "ISConfig timeLapse 120fps as 30fps").
The input surface is told to deliver 120fps but the sensor only produces
---
 .../com/oplus/camera/module/a.smali           | 22 +++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/smali_classes12/com/oplus/camera/module/a.smali b/smali_classes12/com/oplus/camera/module/a.smali
index d878fdda..4c12e49a 100644
--- a/smali_classes12/com/oplus/camera/module/a.smali
+++ b/smali_classes12/com/oplus/camera/module/a.smali
@@ -31910,6 +31910,28 @@
 .method public final ri()Ljava/lang/Double;
     .locals 2
 
+    invoke-virtual {p0}, Lek/u0;->F6()Ljava/lang/String;
+
+    move-result-object v0
+
+    const-string v1, "video_size_8k"
+
+    invoke-virtual {v1, v0}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
+
+    move-result v0
+
+    if-eqz v0, :cond_a12_ri_8k
+
+    const-wide/high16 v0, 0x403e000000000000L
+
+    invoke-static {v0, v1}, Ljava/lang/Double;->valueOf(D)Ljava/lang/Double;
+
+    move-result-object p0
+
+    return-object p0
+
+    :cond_a12_ri_8k
+
     .line 1
     invoke-virtual {p0}, Lcom/oplus/camera/module/a;->r()Z
 
-- 
2.55.0

